src/app/pages/my-request-detail/my-request-detail.component.ts

```diff
@@ -234,201 +379,87 @@ export class MyRequestDetailComponent implements OnInit {
-            return success.item.label;
-          });
-
-          // Remove data element from request queries.
-          return forkJoin({
-            dataQuery: this.removeDataElementFromQuery(labels, this.dataQueryType),
-            cohortQuery: this.removeDataElementFromQuery(labels, this.cohortQueryType),
-            resultMultiple: of(resultMultiple),
-          });
-        }),
-        finalize(() => {
-          this.broadcast.loading({ isLoading: false });
-        })
-      )
-      .subscribe((forkJoinResult) => {
-        this.updateLocalQuery(forkJoinResult.dataQuery);
-        this.updateLocalQuery(forkJoinResult.cohortQuery);
-
-        let success: boolean;
-        let message = '';
-
-        if (items.length === 1) {
-          const singleResult =
-            forkJoinResult.resultMultiple.failures.length === 0
-              ? forkJoinResult.resultMultiple.successes[0]
-              : forkJoinResult.resultMultiple.failures[0];
-          success = singleResult.success;
-
-          if (!singleResult.success) {
-            message = `Removal of data element ${items[0].label} failed. The error message is: ${singleResult.message}`;
-          } else {
-            message = `Data element "${items[0].label}" removed from request "${this.request?.label}".`;
-          }
-        } else {
-          success = forkJoinResult.resultMultiple.failures.length === 0;
-          message = `${forkJoinResult.resultMultiple.successes.length} Data element${
-            forkJoinResult.resultMultiple.successes.length === 1 ? '' : 's'
-          } removed from request "${this.request?.label}".`;
-
-          if (!success) {
-            message += `\r\n${
-              forkJoinResult.resultMultiple.failures.length
-            } Data element${
-              forkJoinResult.resultMultiple.failures.length === 1 ? '' : 's'
-            } caused an error.`;
-          }
-        }
-        this.processRemoveDataElementResponse(success, message);
-      });
-  }
-
-  private updateLocalQuery(newQuery: DataRequestQueryPayload) {
-    if (!newQuery) {
-      return;
-    }
-
-    switch (newQuery.type) {
-      case this.cohortQueryType:
-        this.cohortQuery = newQuery.condition;
-        break;
-      case this.dataQueryType:
-        this.dataQuery = newQuery.condition;
+  onSelectSchema(/* event: SelectionChange*/) {
+    if (this.dataSchemas) {
+      const selectedItemList = this.dataSchemas.filter((item) => item.schema.isSelected);
+      this.selectAllElements = selectedItemList.length === this.dataSchemas.length;
     }
   }
 
-  private removeDataElementFromQuery(
-    dataElementLabels: string[],
-    queryType: DataRequestQueryType
-  ): Observable<DataRequestQueryPayload> {
-    if (!this.request?.id) {
-      return EMPTY;
-    }
+  /**
+   * Methods for general page management
+   */
+  private setBackButtonProperties() {
+    this.backRouterLink = '/requests';
```

Comment:
I have a feeling I removed `backRouterLink` and `backLabel` from a previous PR. They didn't seem necessary because the back link on the page always is the same. Double check this wasn't re-introduced in a rebase.

Reply:
They got half restored in conflict management and my investigations into whether they were required or not was inconclusive. So I put them back to be on the safer side. Now removed.